### PR TITLE
Registry branch

### PR DESCRIPTION
**Pull request by ingenieroariel**
The work was created with funding from NGA via Boundlessgeo.

- Elastic Search and Solr suport
- Docker development and deployment
- Support for adding layers via CSW-T
- Greed behavior for ArcGIS servers is now focused only on the endpoint
- Reusable Django app '/registry/'
- Swagger documented API for catalogs, spatial search, temporal search. Thanks for David Smiley for inspiration.
- Multi Catalog Support
- Maploom UI support for searching on http://localhost/_maploom/
- Documentation in _docs
- Name change to django-registry ahead of it's potential use within GeoNode. Since Hypermap is a bigger project, Ben agreed to call this HHypermap Registry.
- Docker based tests, will support testing against a live solr and elasticsearch in travis.

### CODE REVIEW

**Comment by ingenieroariel** [comment]
@capooti could you comment on the changes when you get a chance?

File: _docs/installation.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+REGISTRY = True # ensure the value is True
+REGISTRYURL = '%s/registry' % SITE_URL.rstrip('/')
+CATALOGLIST = [
+    {
+        "name": "local registry",
+        "url": "%s/hypermap/" % SITE_URL.rstrip("/")
+    },
+]
+SEARCH_ENABLED = True
```
Comment (capooti) [inline _docs/installation.md]: We should read from os.getenv, as in master. I think this happens also for other settings

File: hypermap/aggregator/solr.py [hunk 2]
```diff
@@ -41,19 +44,79 @@ def layer_to_solr(self, layer):
...
         """Clear all indexes in the solr core"""
-        solr_url = settings.SEARCH_URL
+        solr_url = "{0}/solr/{1}".format(SEARCH_URL, catalog)
         solr = pysolr.Solr(solr_url, timeout=60)
         solr.delete(q='*:*')
-        print 'Solr core cleared'
+        LOGGER.debug('Solr core cleared')
+
+    def update_schema(self, catalog="hypermap"):
+        """
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: Excellent. We need a timestamp field, I think this is useful for you as well:
<field name="timestamp" type="date" indexed="true" stored="true" default="NOW" multiValued="false"/>

File: hypermap/aggregator/tasks.py [hunk 3]
```diff
@@ -1,10 +1,34 @@
...
+    SEARCH_ENABLED = False
+    SEARCH_TYPE = None
+    SEARCH_URL = None
+else:
+    SEARCH_ENABLED = True
+    SEARCH_TYPE = REGISTRY_SEARCH_URL.split('+')[0]
+    SEARCH_URL = REGISTRY_SEARCH_URL.split('+')[1]
+
+
+if REGISTRY_LIMIT_LAYERS > 0:
```
Comment (capooti) [inline hypermap/aggregator/tasks.py]: why we need REGISTRY_LIMIT_LAYERS, can't we use just DEBUG_LAYERS_NUMBER?

File: hypermap/aggregator/utils.py [hunk 4]
```diff
@@ -230,68 +276,101 @@ def create_services_from_endpoint(url):
...
-                        esri_service.url,
-                        'ESRI:ArcGIS:MapServer',
-                        esri_service.mapName,
-                        esri_service.description
-                    )
-            services_created.append(service)
+                services_created.append(service)
+
+        # Don't process ImageServer until the following issue has been resolved:
+        # https://github.com/mapproxy/mapproxy/issues/235
```
Comment (capooti) [inline hypermap/aggregator/utils.py]: How about to import them, but not sync them to the search engine for now?

File: hypermap/aggregator/views.py [hunk 5]
```diff
@@ -90,64 +101,77 @@ def index(request):
...
         if 'index' in request.POST:
-            if settings.SKIP_CELERY_TASK:
+            if settings.REGISTRY_SKIP_CELERY:
                 index_service(service)
             else:
                 index_service.delay(service)
 
-    return render(request, 'aggregator/service_detail.html', {'service': service})
+    return render(request, 'aggregator/service_detail.html', {'service': service,
+                                                              'SEARCH_TYPE': SEARCH_TYPE,
```
Comment (capooti) [inline hypermap/aggregator/views.py]: how about to put his in context_processors.py?

Comment (capooti) [inline hypermap/aggregator/views.py]: It is there already in your PR, so you can avoid to pass it to template

File: hypermap/aggregator/utils.py (hunk 4 above)
Comment (ingenieroariel) [inline hypermap/aggregator/utils.py]: Good idea.

**Pull request merged**